resources/todomvc/architecture-examples/preact/src/todo/components/main.jsx

```diff
@@ -0,0 +1,42 @@
+import { useMemo, useCallback } from "preact/hooks";
+import { getCurrentUrl } from "preact-router";
+
+import { Item } from "./item";
+
+export function Main({ todos, dispatch }) {
+    const route = getCurrentUrl();
+
+    const visibleTodos = useMemo(
```

Comment:
It's probably unnecessary to `useMemo` here because that's just 100 items which isn't much. Moreover in this app probably at each action we're doing `todos` changes, so this makes it even more unneeded.

_That said_ it's also pretty common in real-life apps, so I'm inclined to keep it, but with a comment explaining the trade-off. What do you think?